src/staticdata.c

```diff
@@ -890,7 +890,7 @@ static void jl_insert_into_serialization_queue(jl_serializer_state *s, jl_value_
     }
 
     if (immediate) // must be things that can be recursively handled, and valid as type parameters
-        assert(jl_is_immutable(t) || jl_is_typevar(v) || jl_is_symbol(v) || jl_is_svec(v));
+        assert(jl_is_immutable(t) || jl_is_typevar(v) || jl_is_symbol(v) || jl_is_svec(v) || jl_is_module(v));
```

Comment:
Why is this needed?

Reply:
Without it, adding this `precompile` statement to LazyArtifacts causes this assertion to fail:
```julia
precompile(Tuple{typeof(Artifacts._artifact_str), Module, String, Base.SubString{String}, String, Base.Dict{String, Any}, Base.SHA1, Base.BinaryPlatforms.Platform, Val{LazyArtifacts}})
```

since this ends up serializing a Modules as a type parameter

That's valid in principle, but it's still a TODO item here to make sure the serializer handles this correctly.